src/utils/lts/impl/lts.ts

```diff
@@ -1,64 +1,80 @@
-import { injectable } from 'inversify';
-import { ILts, ILtsOptions } from '../interfaces/lts';
+import { inject, injectable } from 'inversify';
+import { ILts, ILtsOptions } from '../interfaces/ILts';
 import { Moment } from 'moment';
 // eslint-disable-next-line @typescript-eslint/quotes
 import moment = require('moment');
+import { memoize } from '../../memoize/memoize';
+import { Fetch, TYPES } from '../../../container/nodeModulesContainer';
+import { ILoggerFactory } from '../../logger';
+import { ILogger } from '../../logger/interfaces/logger';
 
+const NODE_RELEASE_FILE = `https://raw.githubusercontent.com/nodejs/Release/master/schedule.json`;
 const dateFormat = `YYYY-MM-DD`;
 
-interface ILtsVersion {
-  version: string;
+interface IRemoteNodeVersion {
+  start: string;
+  end: string;
+  lts?: string;
+}
+
+type RemoteNodeVersions = Record<string, IRemoteNodeVersion>;
+
+interface INodeVersion {
   from: Moment;
   to: Moment;
+  version: string;
 }
 
-// taken from https://en.wikipedia.org/wiki/Node.js
-const nodesByDate: ILtsVersion[] = [
-  {
-    version: `4`,
-    from: moment.utc(`2015-10-01`, dateFormat),
-    to: moment.utc(`2018-04-30`, dateFormat),
-  },
-  {
-    version: `6`,
-    from: moment.utc(`2016-10-18`, dateFormat),
-    to: moment.utc(`2019-04-30`, dateFormat),
-  },
-  {
-    version: `8`,
-    from: moment.utc(`2017-10-31`, dateFormat),
-    to: moment.utc(`2019-12-31`, dateFormat),
-  },
-  {
-    version: `10`,
-    from: moment.utc(`2018-10-30`, dateFormat),
-    to: moment.utc(`2021-04-01`, dateFormat),
-  },
-  {
-    version: `12`,
-    from: moment.utc(`2019-10-22`, dateFormat),
-    to: moment.utc(`2022-04-01`, dateFormat),
-  },
-  {
-    version: `14`,
-    from: moment.utc(`2020-10-20`, dateFormat),
-    to: moment.utc(`2023-04-30`, dateFormat),
-  },
-];
-
 @injectable()
 export class Lts extends ILts {
-  constructor() {
+  private readonly logger: ILogger;
+  constructor(@inject(TYPES.Fetch) private readonly fetch: Fetch, loggerFactory: ILoggerFactory) {
     super();
+    this.logger = loggerFactory.getLogger(`LTS`);
   }
 
   async resolveLtsVersion({ date }: ILtsOptions): Promise<string[]> {
-    return nodesByDate
-      .filter((lts: ILtsVersion) => {
-        return lts.from.isBefore(date) && lts.to.isAfter(date);
+    this.logger.debug(`Resolving LTS versions from ${date.toJSON()}`);
+    const versions = await this.getAllVersions();
+    const relevantVersions = versions
+      .filter((lts: INodeVersion) => {
+        return lts.from.isSameOrBefore(date) && lts.to.isSameOrAfter(date);
```

Comment:
can be one line: 
`.filter((lts: INodeVersion) =>  lts.from.isSameOrBefore(date) && lts.to.isSameOrAfter(date) )`

Reply:
I'm actually not fund of one liners that are not very short.
I don't mind changing that if we decide to enforce that style from now on